mgmt/src/models/internal/natconfig/table_extend.rs

```diff
@@ -251,4 +284,228 @@ mod tests {
         let mut vni_table = PerVniTable::new();
         add_peering(&mut vni_table, &peering).expect("Failed to build NAT tables");
     }
+
+    #[test]
+    fn test_address_split() {
+        assert_eq!(
+            address_split(&"1.0.0.0/16".into()),
+            IpAddr::V4(Ipv4Addr::from_str("1.0.128.0").unwrap())
+        );
+        assert_eq!(
+            address_split(&"1.0.0.0/17".into()),
+            IpAddr::V4(Ipv4Addr::from_str("1.0.64.0").unwrap())
+        );
+        assert_eq!(
+            address_split(&"1.0.128.0/17".into()),
+            IpAddr::V4(Ipv4Addr::from_str("1.0.192.0").unwrap())
+        );
+        assert_eq!(
+            address_split(&"1.0.0.0/24".into()),
+            IpAddr::V4(Ipv4Addr::from_str("1.0.0.128").unwrap())
+        );
+        assert_eq!(
+            address_split(&"1.0.0.0/31".into()),
+            IpAddr::V4(Ipv4Addr::from_str("1.0.0.1").unwrap())
+        );
+    }
+
+    #[test]
+    #[allow(clippy::too_many_lines)]
+    fn test_collapse_prefix_lists() {
```

Comment:
these tests look fine to me but I really do think you would have an easier time with bolero.  Won't block PR on it tho :)

Reply:
I was more or less expecting this comment :) But I'm not sure how to do it (in particular, how to validate that the output for non-deterministic inputs will be as expected, without running the same algo we're trying to evaluate). We probably need to talk about it at some point.